internal/connectors/plugins/public/moneycorp/accounts.go

```diff
@@ -37,42 +39,22 @@ func (p Plugin) fetchNextAccounts(ctx context.Context, req models.FetchNextAccou
 		if err != nil {
 			return models.FetchNextAccountsResponse{}, err
 		}
-
 		if len(pagedAccounts) == 0 {
 			break
 		}
 
-		for _, account := range pagedAccounts {
-			if account.ID <= oldState.LastIDCreated {
-				continue
-			}
-
-			raw, err := json.Marshal(account)
-			if err != nil {
-				return models.FetchNextAccountsResponse{}, err
-			}
-
-			accounts = append(accounts, models.PSPAccount{
-				Reference: account.ID,
-				// Moneycorp does not send the opening date of the account
-				CreatedAt: time.Now().UTC(),
-				Name:      &account.Attributes.AccountName,
-				Raw:       raw,
-			})
-
-			newState.LastIDCreated = account.ID
-
-			if len(accounts) >= req.PageSize {
-				break
-			}
+		accounts, err = toPSPAccounts(oldState.LastIDCreated, accounts, pagedAccounts)
+		if err != nil {
+			return models.FetchNextAccountsResponse{}, err
 		}
-
-		if len(pagedAccounts) < req.PageSize {
-			break
+		if len(accounts) == 0 {
+			continue
 		}
+		newState.LastIDCreated = accounts[len(accounts)-1].Reference
```

Comment:
I would have put the LastIDCreated out of the loop here, because the should fetch more function is truncated the accounts when > pageSize, we will miss some accounts at the next fetching

Reply:
I made the pageSize sent to the client dynamic so that we don't exceed the preallocated size of `accounts`. The new ShouldFetchMore also no longer adjusts the size of the slice so it should be fine where it is now.